python/ray/exceptions.py

```diff
@@ -129,7 +129,17 @@ class ObjectStoreFullError(RayError):
     This is raised if the attempt to store the object fails
     because the object store is full even after multiple retries.
     """
-    pass
+
+    def __str__(self):
+        return (
+            "The local object store is full of objects that are still in scope"
+            " and cannot be evicted. Try increasing the object store memory "
+            "available with ray.init(object_store_memory=<bytes>). "
+            "You can also try setting an option to fallback to LRU eviction "
+            "when the object store is full by calling ray.init("
+            "_internal_config=json.dumps({\"object_pinning_enabled\": 0})). "
```

Comment:
We shouldn't ship with such an ugly message, how about `ray.init(lru_evict=True)` or similar.